src/MakingSense.AspNetCore.Documentation/DocumentationMiddleware.cs

```diff
@@ -147,14 +163,32 @@ private string GetFilePath(string subpath, out bool isDirectory)
 				: subpath;
 		}
 
-		private bool TryGetDocumentsSubpath(HttpRequest request, out PathString subpath)
+		private bool TryGetDocumentsSubpathAndLanguage(HttpRequest request, out PathString subpath, out string lang)
 		{
+			lang = null;
+
 			if (!request.Path.StartsWithSegments(_options.RequestPath, out subpath))
 			{
 				return false;
 			}
 
-			subpath = _options.FileProviderSubPath.Add(subpath);
+			string langPattern = @"^(\/[a-zA-Z]{2})(\/.*){0,1}$"; //eg. /en or /en/somedir/somefile
+			Regex r = new Regex(langPattern, RegexOptions.IgnoreCase);
+
+			var match = r.Match(subpath);
+
+			if (match.Success)
+			{
+				lang = match.Groups[1].Value.TrimStart('/');
+				if(match.Groups.Count == 3)
+				{
+					subpath = _options.FileProviderSubPath.Add(match.Groups[2].Value);
+				}
```

Comment:
I think that there is a missed scenario here, when _success_ but _Groups.Count_ is different than _3_